modules/openxr/openxr_interface.cpp

```diff
@@ -690,6 +690,20 @@ bool OpenXRInterface::supports_play_area_mode(XRInterface::PlayAreaMode p_mode)
 }
 
 XRInterface::PlayAreaMode OpenXRInterface::get_play_area_mode() const {
+	if (!openxr_api) {
```

Comment:
Don't forget to also implement `supports_play_area_mode` and `set_play_area_mode`

Reply:
Do we want to support changing the reference space mid-game? Right now it takes this value from a project setting that's only used during initialization.

Reply:
I wonder if we can, forgot that we do this during init. If we need to tear down a session and rebuild it, it will be overkill and it indeed makes more sense that `set_play_area_mode` gives and error after OpenXR is initialised. But if it can change on the fly that would be cool.

Imagine switching between `LOCAL` and `LOCAL_FLOOR` as you enter/exit a vehicle

Reply:
I think we can. But, personally, I'd be fine with this only being allowed during init (or like you suggest, giving an error if set when OpenXR is already initialized), at least until we hit a compelling use case that really needs it. I worry that giving developers this power would cause more problems than it would solve. :-)

Reply:
Agreed :)

Reply:
@dsnopek @BastiaanOlij I ran into this while working on the XR editor and I'd suggest removing the restriction (if possible) that the api can only be used prior to initialization.

The project setting applies to the entire project, which doesn't provide much granularity. And trying to use this api prior to the `XRInterface` being initialized is practically impossible since by the time `gdscript` can get a handle to the `XRInterface` instance, _it's already initialized_.

Reply:
Hm. In a normal project, you'd set it in the "Project Settings" so that it's at the value you want during initialization, however, I can see how that would be an issue in the XR editor.

From a technical perspective, I think we are able to change reference space after OpenXR is initialized, and so we could remove the restriction. However, I don't know what the experience would be in the headset - I can do some experimentation. In the worst case (like if it's really weird in headset), I guess we can recommend in the docs that developers are careful with this function and maybe black out the camera before doing it or re-center on the HMD after doing it, or something like that?

Reply:
We can also recommend switching the reference space only during transition from one scene to another.

For now for the XR editor, I'm creating an editor override for the reference space project setting in https://github.com/m4gr3d/godot/commit/6dd167a3bcc88dc789e36fdc6b9d9b967b6d7321.

Ideally, it should be possible to update the reference space while the editor is being initialized in https://github.com/m4gr3d/godot/blob/6dd167a3bcc88dc789e36fdc6b9d9b967b6d7321/modules/vr_editor/vr_editor.cpp#L44

Reply:
@m4gr3d Here's PR https://github.com/godotengine/godot/pull/87745 to allow changing the reference space even after OpenXR has been initialized